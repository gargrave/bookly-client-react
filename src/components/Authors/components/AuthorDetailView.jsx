import React from 'react'
import { func, number, shape, string } from 'prop-types'

const AuthorDetailView = (props) => {
  const { author, handleEditClick, handleBackClick } = props
  return (
    <div className="author-detail-view">
      <h2>Author Detail</h2>
      <p className="author-name">
        <strong>Name:</strong> {author.firstName} {author.lastName}
      </p>
      <p className="author-added-on">
        <strong>Added on:</strong> {author.createdAt}
      </p>
      <p className="author-updated-on">
        <strong>Updated on:</strong> {author.updatedAt}
      </p>
      <button type="button" className="author-edit-button" onClick={handleEditClick}>
        Edit
      </button>&nbsp;
      <button type="button" className="button-info cancel-button" onClick={handleBackClick}>
        Back
      </button>
    </div>
  )
}

AuthorDetailView.propTypes = {
  author: shape({
    id: number,
    firstName: string,
    lastName: string,
    createdAt: string,
    updatedAt: string,
  }),
  handleEditClick: func.isRequired,
  handleBackClick: func.isRequired,
}

export default AuthorDetailView
