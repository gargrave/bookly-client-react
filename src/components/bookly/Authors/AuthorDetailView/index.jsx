import React from 'react'
import { func, number, shape, string } from 'prop-types'
import { format } from 'date-fns'

import { buildClasses } from '@/utils/cssHelpers'

import Button from '@/components/common/Button'
import Card from '@/components/common/Card'

const AuthorDetailView = (props) => {
  const {
    author,
    onEditClick,
    onBackClick,
  } = props
  return (
    <div className={buildClasses('author-detail-view')}>
      <Card
        classes={['detail-card', 'author-detail-card']}
        hoverable={false}
        title={`${author.firstName} ${author.lastName}`}>

        <hr/>
        <p className={buildClasses('card-text')}>
          <strong>Added:</strong> {format(author.createdAt, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>
        <p className={buildClasses('card-text')}>
          <strong>Updated:</strong> {format(author.updatedAt, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>

        <hr/>
        <Button
          onClick={onEditClick}
          text="Edit"
          type="info" />
        <Button
          onClick={onBackClick}
          text="Back"
          type="light" />
      </Card>
    </div>
  )
}

AuthorDetailView.propTypes = {
  author: shape({
    id: number,
    firstName: string,
    lastName: string,
    createdAt: string,
    updatedAt: string,
  }),
  onBackClick: func.isRequired,
  onEditClick: func.isRequired,
}

export default AuthorDetailView
