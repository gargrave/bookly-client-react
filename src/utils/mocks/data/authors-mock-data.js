export const authorsMockData = [
  {
    id: 8,
    firstName: 'Aldous',
    lastName: 'Huxley',
    createdAt: '2017-07-01T01:16:43.881Z',
    updatedAt: '2017-07-01T01:16:43.881Z'
  },
  {
    id: 51,
    firstName: 'Kurt',
    lastName: 'Vonnegut',
    createdAt: '2017-07-06T22:02:11.504Z',
    updatedAt: '2017-07-06T22:02:11.504Z'
  },
  {
    id: 59,
    firstName: 'George R. R.',
    lastName: 'Martin',
    createdAt: '2017-08-17T20:47:35.232Z',
    updatedAt: '2017-08-17T20:47:35.232Z'
  },
  {
    id: 52,
    firstName: 'Tom',
    lastName: 'Robbins',
    createdAt: '2017-07-06T22:02:17.051Z',
    updatedAt: '2017-08-17T21:27:35.885Z'
  }
]
