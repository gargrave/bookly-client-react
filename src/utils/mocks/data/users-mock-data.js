export const usersMockData = [
  {
    id: 1,
    email: 'mockuser1@gmail.com',
    createdAt: '2017-06-30T19:52:54.644Z',
    updatedAt: '2017-06-30T19:52:54.644Z',
    lastLogin: '2017-07-01T01:11:03.942Z',
    verified: true,
  },
];
