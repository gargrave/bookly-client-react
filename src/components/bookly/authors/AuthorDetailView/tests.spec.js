import React from 'react'
import { shallow } from 'enzyme'

import { authorsMockData } from '@/utils/mocks/data/authors-mock-data'
import AuthorDetailView from './'

describe('AuthorDetailView', () => {
  let props
  let component

  beforeEach(() => {
    props = {
      author: Object.create(authorsMockData[0]),
      handleEditClick: jest.fn(),
      handleBackClick: jest.fn(),
    }

    component = shallow(<AuthorDetailView {...props} />)
  })

  test('renders correctly', () => {
    const element = component.find('.author-detail-view')
    expect(component).toMatchSnapshot()
    expect(element.length).toEqual(1)
  })

  test("renders the author's name", () => {
    const text = component.find('.author-name').text()
    expect(text.indexOf(`${props.author.firstName} ${props.author.lastName}`)).not.toEqual(-1)
  })

  test("renders the author's 'added on' date", () => {
    const text = component.find('.author-added-on').text()
    expect(text.indexOf(props.author.createdAt)).not.toEqual(-1)
  })

  test("renders the author's 'updated on' date", () => {
    const text = component.find('.author-updated-on').text()
    expect(text.indexOf(props.author.updatedAt)).not.toEqual(-1)
  })

  test('calls the "handleEditClick" callback when the button is clicked', () => {
    expect(props.handleEditClick.mock.calls.length).toEqual(0)
    component.find('.author-edit-button').simulate('click')
    expect(props.handleEditClick.mock.calls.length).toEqual(1)
  })

  test('calls the "handleBackClick" callback when the button is clicked', () => {
    expect(props.handleBackClick.mock.calls.length).toEqual(0)
    component.find('.cancel-button').simulate('click')
    expect(props.handleBackClick.mock.calls.length).toEqual(1)
  })
})
