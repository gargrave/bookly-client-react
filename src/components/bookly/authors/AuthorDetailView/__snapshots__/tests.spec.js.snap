// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDetailView renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorDetailView
    author={Object {}}
    onBackClick={[Function]}
    onEditClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Author Detail
        </h2>,
        <p
          className="author-name"
        >
          <strong>
            Name:
          </strong>
           
          Aldous
           
          Huxley
        </p>,
        <p
          className="author-added-on"
        >
          <strong>
            Added on:
          </strong>
           
          2017-07-01T01:16:43.881Z
        </p>,
        <p
          className="author-updated-on"
        >
          <strong>
            Updated on:
          </strong>
           
          2017-07-01T01:16:43.881Z
        </p>,
        <Button
          onClick={[Function]}
          text="Edit"
          type="info"
        />,
        <Button
          onClick={[Function]}
          text="Back"
          type="light"
        />,
      ],
      "className": "author-detail-view",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Author Detail",
        },
        "ref": null,
        "rendered": "Author Detail",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Name:
            </strong>,
            " ",
            "Aldous",
            " ",
            "Huxley",
          ],
          "className": "author-name",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name:",
            },
            "ref": null,
            "rendered": "Name:",
            "type": "strong",
          },
          " ",
          "Aldous",
          " ",
          "Huxley",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Added on:
            </strong>,
            " ",
            "2017-07-01T01:16:43.881Z",
          ],
          "className": "author-added-on",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Added on:",
            },
            "ref": null,
            "rendered": "Added on:",
            "type": "strong",
          },
          " ",
          "2017-07-01T01:16:43.881Z",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Updated on:
            </strong>,
            " ",
            "2017-07-01T01:16:43.881Z",
          ],
          "className": "author-updated-on",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Updated on:",
            },
            "ref": null,
            "rendered": "Updated on:",
            "type": "strong",
          },
          " ",
          "2017-07-01T01:16:43.881Z",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "text": "Edit",
          "type": "info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "text": "Back",
          "type": "light",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Author Detail
          </h2>,
          <p
            className="author-name"
          >
            <strong>
              Name:
            </strong>
             
            Aldous
             
            Huxley
          </p>,
          <p
            className="author-added-on"
          >
            <strong>
              Added on:
            </strong>
             
            2017-07-01T01:16:43.881Z
          </p>,
          <p
            className="author-updated-on"
          >
            <strong>
              Updated on:
            </strong>
             
            2017-07-01T01:16:43.881Z
          </p>,
          <Button
            onClick={[Function]}
            text="Edit"
            type="info"
          />,
          <Button
            onClick={[Function]}
            text="Back"
            type="light"
          />,
        ],
        "className": "author-detail-view",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Author Detail",
          },
          "ref": null,
          "rendered": "Author Detail",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Name:
              </strong>,
              " ",
              "Aldous",
              " ",
              "Huxley",
            ],
            "className": "author-name",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name:",
              },
              "ref": null,
              "rendered": "Name:",
              "type": "strong",
            },
            " ",
            "Aldous",
            " ",
            "Huxley",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Added on:
              </strong>,
              " ",
              "2017-07-01T01:16:43.881Z",
            ],
            "className": "author-added-on",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Added on:",
              },
              "ref": null,
              "rendered": "Added on:",
              "type": "strong",
            },
            " ",
            "2017-07-01T01:16:43.881Z",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Updated on:
              </strong>,
              " ",
              "2017-07-01T01:16:43.881Z",
            ],
            "className": "author-updated-on",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Updated on:",
              },
              "ref": null,
              "rendered": "Updated on:",
              "type": "strong",
            },
            " ",
            "2017-07-01T01:16:43.881Z",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "text": "Edit",
            "type": "info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "text": "Back",
            "type": "light",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
